nbconvert/exporters/export.py

```diff
@@ -165,30 +183,30 @@ def export_by_name(format_name, nb, **kw):
         Name of the template style to export to.
     """
     
-    exporter = get_exporter(name)
-    function_name = "export_" + format_name.lower()
-    
-    if function_name in globals():
-        return globals()[function_name](nb, **kw)
-    else:
-        exporter = get_exporter(name)
-        
-        raise ExporterNameError("Exporter for `%s` not found" % function_name)
+    warnings.warn("export_by_name is deprecated since nbconvert 5.0", DeprecationWarning, stacklevel=3)
```

Comment:
Also, I think it's a good idea for a deprecation warning to say what the suggested replacement is, i.e.

```python
export(get_exporter(format_name), nb, **kw))
```